hive/agents/td3.py

```diff
@@ -0,0 +1,424 @@
+        self._target_actor.train()
+        self._target_critic.train()
+
+    def eval(self):
+        """Changes the agent to evaluation mode."""
+        super().eval()
+        self._actor.eval()
+        self._critic.eval()
+        self._target_actor.eval()
+        self._target_critic.eval()
+
+    def scale_action(self, actions):
+        """Scales actions to [-1, 1]."""
+        if self._scale_actions:
+            return ((actions - self._action_min) / self._action_scaling) - 1.0
+        else:
+            return actions
+
+    def unscale_actions(self, actions):
+        """Unscales actions from [-1, 1] to expected scale."""
+        if self._scale_actions:
+            return ((actions + 1.0) * self._action_scaling) + self._action_min
+        else:
+            return actions
+
+    def preprocess_update_info(self, update_info):
+        """Preprocesses the :obj:`update_info` before it goes into the replay buffer.
+        Scales the action to [-1, 1].
+
+        Args:
+            update_info: Contains the information from the current timestep that the
+                agent should use to update itself.
+        """
+        if self._reward_clip is not None:
+            update_info["reward"] = np.clip(
+                update_info["reward"], -self._reward_clip, self._reward_clip
+            )
+        preprocessed_update_info = {
+            "observation": update_info["observation"],
+            "action": self.scale_action(update_info["action"]),
+            "reward": update_info["reward"],
+            "done": update_info["done"],
+        }
+        if "agent_id" in update_info:
+            preprocessed_update_info["agent_id"] = int(update_info["agent_id"])
+
+        return preprocessed_update_info
+
+    def preprocess_update_batch(self, batch):
+        """Preprocess the batch sampled from the replay buffer.
+
+        Args:
+            batch: Batch sampled from the replay buffer for the current update.
+
+        Returns:
+            (tuple):
+                - (tuple) Inputs used to calculate current state values.
+                - (tuple) Inputs used to calculate next state values
+                - Preprocessed batch.
+        """
+        for key in batch:
+            batch[key] = torch.tensor(batch[key], device=self._device)
+        return (batch["observation"],), (batch["next_observation"],), batch
+
+    @torch.no_grad()
+    def act(self, observation):
+        """Returns the action for the agent. If in training mode, adds noise with
+        standard deviation :py:obj:`self._action_noise`.
+
+        Args:
+            observation: The current observation.
+        """
+
+        # Calculate action and add noise if training.
+        observation = torch.tensor(
+            np.expand_dims(observation, axis=0), device=self._device
+        ).float()
+        action = self._actor(observation)
+        if self._training:
+            noise = torch.randn_like(action, requires_grad=False) * self._action_noise
```

Comment:
No changes but a couple of comments.
1. File structure and naming has to be changed. I guess that can be handled  in a later PR.
2. Feature request here! Add an exploration class that can handle different exploration techniques? Uniform, gaussian, boltzmann

Rest everything looks good!